playwright/src/main/java/com/deque/html/axecore/playwright/AxeBuilder.java

```diff
@@ -0,0 +1,406 @@
+   *
+   * @param selector ArrayList of Strings
+   * @return this
+   */
+  public AxeBuilder exclude(List<String> selector) {
+    this.context.setExclude(selector);
+    return this;
+  }
+
+  /**
+   * Provide options how configure how axe-core operates Run all rules corresponding to one of the
+   * accessibility standards Run all rules defined in the system, except for the list of rules
+   * specified Run a specific set of rules provided as a list of rule ids
+   *
+   * @param axeRunOptions axe-core options
+   * @return this
+   */
+  public AxeBuilder options(AxeRunOptions axeRunOptions) {
+    this.options = axeRunOptions;
+    return this;
+  }
+
+  /**
+   * Limit amount of rules to be executed during analysis
+   *
+   * @param rules ArrayList of rules to be executed
+   * @return this
+   */
+  public AxeBuilder withRules(List<String> rules) {
+    if (!rules.isEmpty()) {
+      AxeRunOnlyOptions runOnlyOptions = new AxeRunOnlyOptions();
+      runOnlyOptions.setType("rule");
+      runOnlyOptions.setValues(rules);
+      this.options.setRunOnly(runOnlyOptions);
+    } else {
+      throw new RuntimeException(
+          "withRules list cannot be empty. "
+              + "Please see: https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md#rule-descriptions");
+    }
+    return this;
+  }
+
+  /**
+   * Limit the tags (accessibility standards) to be executed during analysis
+   *
+   * @param tags ArrayList of tags to executed
+   * @return this
+   * @see <a
+   *     href="https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#axe-core-tags">axe-core
+   *     Tags</a>
+   */
+  public AxeBuilder withTags(List<String> tags) {
+    if (!tags.isEmpty()) {
+      AxeRunOnlyOptions runOnlyOptions = new AxeRunOnlyOptions();
+      runOnlyOptions.setType("tag");
+      runOnlyOptions.setValues(tags);
+      this.options.setRunOnly(runOnlyOptions);
+    } else {
+      throw new RuntimeException(
+          "withTags list cannot be empty. "
+              + "Please see: https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#axe-core-tags");
+    }
+    return this;
+  }
+
+  /**
+   * Disable rules to be executed during analysis
+   *
+   * @param rules ArrayList of rules to be disabled
+   * @return this
+   */
+  public AxeBuilder disableRules(List<String> rules) {
+    if (!rules.isEmpty()) {
+      Map<String, AxeRuleOptions> disableRulesMap = new HashMap<>();
+      rules.forEach(
+          (rule) -> {
+            AxeRuleOptions axeRuleOptions = new AxeRuleOptions();
+            axeRuleOptions.setEnabled(false);
+            disableRulesMap.put(rule, axeRuleOptions);
+          });
+      this.options.setRules(disableRulesMap);
+    } else {
+      throw new RuntimeException(
+          "disableRules list cannot be empty. "
+              + "Please see: https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md#rule-descriptions");
+    }
```

Comment:
Nit - flip the `if` so we can exit early. Doing so keeps the happy-path closer indented to the left, which tends to make for easier to read code.

```suggestion
     if (rules.isEmpty()) {
      throw new RuntimeException(
          "disableRules list cannot be empty. "
              + "Please see: https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md#rule-descriptions");
    }
    Map<String, AxeRuleOptions> disableRulesMap = new HashMap<>();
    rules.forEach(
        (rule) -> {
          AxeRuleOptions axeRuleOptions = new AxeRuleOptions();
          axeRuleOptions.setEnabled(false);
          disableRulesMap.put(rule, axeRuleOptions);
        });
    this.options.setRules(disableRulesMap);
```

Nice doc on this- https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

Reply:
Oh, good article - I agree will update shortly